ci-operator/step-registry/baremetalds/e2e/upgrade/upgrade/baremetalds-e2e-upgrade-upgrade-commands.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+openshift-tests run-upgrade --to-image "${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}" --dry-run all
```

Comment:
This unfortunately cannot work, since the `openshift-tests` binary must be copied on the remote Packet instance where the cluster is currently running. In addition, tests results must be copied back so that they could be published in the job dashboard by the CI.
The `baremetalds-e2e-test` it's already taking care of all these task, it would be sufficient to make the `openshift-tests` command here (https://github.com/openshift/release/blob/8cb0233ac6d7a0a4c3b445fae410610ecd7d72b3/ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh#L41) conditional

Reply:
@andfasano From what I've seen, the `openshift-tests` binary is copied to the remote Packet instance on the `baremetal-e2e-test` step.

See: https://github.com/openshift/release/blob/8cb0233ac6d7a0a4c3b445fae410610ecd7d72b3/ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh#L33

So therefore, the `openshift-tests` binary is available for the upgrade step is the test step is executed beforehand.
We can either merge those 2 steps (as you've mentioned) or make sure the upgrade step handles copies the 'openshift-tests` binary itself.

Reply:
Perhaps we could mimic vSphere approach here. There is a permanent cluster running on vSphere, which reuses the platform to create more clusters. In that case ssh/scp is not necessary as  permanent cluster can reach the new one directly via KUBECONFIG.
WDYT?

Reply:
I'd prefer to merge it within the existing and already working `baremetalds-e2e-test-command`. Changes we'll be really minimal, and the step it's already taking care of all the rest. In addition, this will let us quickly go on and focusing on the `openshift-tests` upgrade tests execution itself, which could present some additional difficulties (hopefully not) for the baremetal platform.